commons-vfs2/src/main/java/org/apache/commons/vfs2/provider/UriParser.java

```diff
@@ -117,7 +118,10 @@ private boolean readSeparator() {
             }
             final String sub = path.substring(cursor, cursor + 3);
             if (sub.equals(URLENCODED_SLASH_LC) || sub.equals(URLENCODED_SLASH_UC)) {
-                cursor += 3;
+                path.setCharAt(cursor, SEPARATOR_CHAR);
+                path.delete(cursor + 1, cursor + 3);

```

Comment:
Should we use constants instead of magic numbers here? This would help maintenance in the future IMO.

Reply:
This pattern of `path.setCharAt` and `path.delete(cursor + 1, cursor + 3);` is already used in `canonicalizePath()` and `decode()` methods.
There is in commons-vfs a checkstyle or pmd rule for magic numbers and it starts with number 4 and above.